Sources/Models/AppResponse.swift

```diff
@@ -29,35 +29,84 @@ public struct AppResponse: Decodable {
-            }
-            if let wrapped = try? BetaLicenseAgreement(from: decoder) {
-                self = .betaLicenseAgreement(wrapped)
-                return
-            }
-            if let wrapped = try? BetaAppReviewDetail(from: decoder) {
-                self = .betaAppReviewDetail(wrapped)
-                return
-            }
+            enum TypeCodingKeys: String, CodingKey { case type }
             
-            throw DecodingError.typeMismatch(
-                Included.self,
-                DecodingError.Context(codingPath: [], debugDescription: "Not convertable to \(Included.self)")
-            )
+            switch try decoder.container(keyedBy: TypeCodingKeys.self).decode(String.self, forKey: .type) {
+            case "betaGroups":
+                self = try .betaGroup(BetaGroup(from: decoder))
+            case "prereleaseVersions":
+                self = try .prereleaseVersion(PrereleaseVersion(from: decoder))
+            case "betaAppLocalizations":
+                self = try .betaAppLocalization(BetaAppLocalization(from: decoder))
+            case "builds":
+                self = try .build(Build(from: decoder))
+            case "betaLicenseAgreements":
+                self = try .betaLicenseAgreement(BetaLicenseAgreement(from: decoder))
+            case "betaAppReviewDetails":
+                self = try .betaAppReviewDetail(BetaAppReviewDetail(from: decoder))
+            default:
+                throw DecodingError.typeMismatch(
+                    Included.self,
+                    DecodingError.Context(codingPath: [], debugDescription: "Not convertable to \(Included.self)")
+                )
+            }
+        }
+    }
+}
+
+extension AppResponse.Included {
+    
+    public var betaGroup: BetaGroup? {
+        switch self {
+        case .betaGroup(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var prereleaseVersion: PrereleaseVersion? {
+        switch self {
+        case .prereleaseVersion(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var betaAppLocalization: BetaAppLocalization? {
+        switch self {
+        case .betaAppLocalization(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var build: Build? {
+        switch self {
+        case .build(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var betaLicenseAgreement: BetaLicenseAgreement? {
+        switch self {
+        case .betaLicenseAgreement(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var betaAppReviewDetail: BetaAppReviewDetail? {
+        switch self {
+        case .betaAppReviewDetail(let value):
+            return value
+        default:
+            return nil
```

Comment:
This adds quite some overhead. Do you think it's possible to make this more efficient in less duplicate code?

Reply:
I could not come up with a nicer alternative for now, I'll remove those and maybe think about it later :)